hedera-node/src/main/java/com/hedera/services/contracts/sources/TxnAwareEvmSigsVerifier.java

```diff
@@ -189,20 +166,27 @@ BiPredicate<JKey, TransactionSignature> validityTestFor(
 				final var controllingContract = aliases.currentAddress(controllingId);
 				return !isDelegateCall && controllingContract.equals(activeContract);
 			} else {
-				/* Otherwise apply the standard cryptographic validity test */
+				// Otherwise, apply the standard cryptographic validity test
 				return cryptoValidity.test(key, sig);
 			}
 		};
 	}
 
 	private Optional<JKey> receiverSigKeyIfAnyOf(final AccountID id, final WorldLedgers worldLedgers) {
-		final var merkleAccount = worldLedgers.accounts() != null ?
-				Optional.ofNullable(worldLedgers.accounts().getImmutableRef(id)) :
-				Optional.empty();
-
-		return merkleAccount
-				.filter(account -> !((MerkleAccount)account).isSmartContract())
-				.filter(account -> ((MerkleAccount)account).isReceiverSigRequired())
-				.map(account -> ((MerkleAccount)account).getAccountKey());
+		final var accounts = worldLedgers.accounts();
+		if (accounts == null) {
+			// This must be a static call, hence cannot contain value and cannot require a signature
+			return Optional.empty();
+		}
+		return isReceiverSigExempt(id, accounts) ? Optional.empty() : Optional.ofNullable((JKey) accounts.get(id, KEY));
+	}
+
+	private boolean isReceiverSigExempt(
+			final AccountID id,
+			final TransactionalLedger<AccountID, AccountProperty, MerkleAccount> accounts
+	) {
+		return !accounts.contains(id)
+				|| (boolean) accounts.get(id, IS_SMART_CONTRACT)
+				|| !(boolean) accounts.get(id, IS_RECEIVER_SIG_REQUIRED);
```

Comment:
we can remove this check `accounts.get(id, IS_RECEIVER_SIG_REQUIRED)` as contracts will not have receiver sig req set.

Reply:
Removed.